spec/integration/post_index_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.feature 'Post Index', type: :feature do
+  let(:user) { User.create(name: 'Tom', photo: 'https://www.kasandbox.org/programming-images/avatars/leaf-blue.png', bio: 'He is a good programmer') }
+  let!(:post) { Post.create(author: user, title: "first post's title", text: 'first text') }
+  let!(:comment) { Comment.create(post: post, text: 'first comment') }
+  let!(:like1) { Like.create(post: post) }
+
+  scenario "see user's profile picture, username, number of posts and interactions" do
+    visit user_posts_path(user, post)
+    expect(page).to have_content('Tom')
+    expect(page).to have_css("img[alt='Tom']", count: 1)
+    expect(page).to have_content('post')
+  end
+
+  scenario "see some of the post's title, body and first comments" do
+    visit user_posts_path(user, post, comment)
+    expect(page).to_not have_content('first comment')
+  end
+
+  scenario 'see a section for pagination if there are more posts than fit on the view' do
+    visit user_posts_path(user)
+    expect(page).to have_content('Pagination')
+  end
+
+  scenario "clicking on a post redirects me to that post's show page" do
+    visit user_posts_path(user, post)
+    expect(page).to have_current_path(user_posts_path(user, post))
+  end
```

Comment:
While you did a good job working on the requirements your tests are not passing and you missed out some test you have a total of 15 examples but the required of features to test are about 32 or so. kindly fix your failing tests and and more tests.

![Screenshot 2023-11-10 at 10 54 05 AM](https://github.com/2004-okari/BLOG_APP/assets/79506105/39d72769-e10e-410c-8014-76aa060afec0)

It would be nice to write your tests based on the following requirements:

2. Has featured tests for User index page:

Spec file should only test User index page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for the redirect
Tests that asserts you can see the username of all other users.
Tests that asserts you can see the profile picture for each user.
Tests that asserts you can see the number of posts each user has written.
Tests that redirects to the user's show page when you click on a user.


3. Has featured tests for User show page:

Spec file should only test User show page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for each redirect
Tests that asserts you can see the user's profile picture.
Tests that asserts you can see the user's username.
Tests that asserts you can see the number of posts the user has written.
Tests that asserts you can see the user's bio.
Tests that asserts you can see the user's first 3 posts.
Tests that asserts you can see a button that lets me view all of a user's posts.
Tests that redirects to the post's show page when you click on a user's post.
Tests that redirects to the user's post's index page when you click on a see all posts.


4. Has featured tests for User post index page:

Spec file should only test User post index page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for the redirect
Tests that asserts you can see the user's profile picture.
Tests that asserts you can see the user's username.
Tests that asserts you can see the number of posts the user has written.
Tests that asserts you can see a post's title.
Tests that asserts you can see some of the post's body.
Tests that asserts you can see the first comments on a post.
Tests that asserts you can see how many comments a post has.
Tests that asserts you can see how many likes a post has.
Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
Tests that redirects to the post's show page when you click on a post.


5. Has featured tests for User post show page:

Spec file should only test User show page. It can be only one spec testing all off the things below or more.
Tests that asserts you can see the post's title.
Tests that asserts you can see who wrote the post.
Tests that asserts you can see how many comments it has.
Tests that asserts you can see how many likes it has.
Tests that asserts you can see the post body.
Tests that asserts you can see the username of each commentor.
Tests that asserts you can see the comment each commentor left.

Reply:
- Combined some tests together but have all the required testa are there.
- Fix the tests
<img width="898" alt="Screenshot 2023-11-10 at 5 01 37 PM" src="https://github.com/2004-okari/BLOG_APP/assets/137653639/d6bbdd6e-e064-4026-a82f-6cff1e062a31">